core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -80,27 +85,49 @@ oppia.factory('playerTranscriptService', ['$log', function($log) {
       lastCard.destStateName = newDestStateName;
     },
     addNewAnswer: function(answer) {
-      var pairs = transcript[transcript.length - 1].answerFeedbackPairs;
+      var pairs = transcript[transcript.length - 1].inputResponsePairs;
       if (pairs.length > 0 &&
           pairs[pairs.length - 1].oppiaFeedbackHtml === null) {
         throw Error(
           'Trying to add an answer before the feedback for the previous ' +
           'answer has been received.',
           transcript);
       }
-      transcript[transcript.length - 1].answerFeedbackPairs.push({
+      transcript[transcript.length - 1].inputResponsePairs.push({
         learnerAnswer: answer,
         oppiaFeedbackHtml: null
       });
     },
+    addNewHintRequest: function(hintRequest) {
```

Comment:
No, there's still a race here. How about this for now -- if user submits answer/hint while there's a request pending, just return at the outset, don't process it. Make sure that if they then click the relevant button again subsequently (after the response comes back), everything proceeds as normal.

The above seems least bad to me, and you don't need to make any changes in the frontend for now. We need to find a proper solution for this and handle it in a separate PR after the release (please discuss with your mentors). Probably the easiest long-term solution is to just disable the submit/hint buttons while a request is in process.